doc/tunnel/SAI-Proposal-EVPN-Multihoming.md

```diff
@@ -0,0 +1,616 @@
+...
+    /** Next hop group is for bridge port */
+    SAI_NEXT_HOP_GROUP_TYPE_BRIDGE_PORT,
+
+} sai_next_hop_group_type_t;
+```
+
+A new nexthop type which will be part of groups of the above group type.
+
+```
+typedef enum _sai_next_hop_type_t
+...
+    /** Next hop group is for bridge port */
+    SAI_NEXT_HOP_TYPE_BRIDGE_PORT,
+} sai_next_hop_type_t;
+```
+
+The above nexthop type will need ATTR_TUNNEL_ID as well as ATTR_IP as mandatory parameters on creation.
+
+```
+typedef enum _sai_next_hop_attr_t
+...
+    /**
+     * @brief Next hop entry IPv4 address
+     *
+     * @type sai_ip_address_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     * @condition SAI_NEXT_HOP_ATTR_TYPE == SAI_NEXT_HOP_TYPE_IP or SAI_NEXT_HOP_ATTR_TYPE == SAI_NEXT_HOP_TYPE_MPLS or SAI_NEXT_HOP_ATTR_TYPE == SAI_NEXT_HOP_TYPE_TUNNEL_ENCAP or SAI_NEXT_HOP_ATTR_TYPE == SAI_NEXT_HOP_TYPE_BRIDGE_PORT
+     */
+    SAI_NEXT_HOP_ATTR_IP,
+
+
+    /**
+     * @brief Next hop entry tunnel-id
+     *
+     * @type sai_object_id_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     * @objects SAI_OBJECT_TYPE_TUNNEL
+     * @condition SAI_NEXT_HOP_ATTR_TYPE == SAI_NEXT_HOP_TYPE_TUNNEL_ENCAP or SAI_NEXT_HOP_ATTR_TYPE == SAI_NEXT_HOP_TYPE_SRV6_SIDLIST or SAI_NEXT_HOP_ATTR_TYPE == SAI_NEXT_HOP_TYPE_BRIDGE_PORT
+     */
+    SAI_NEXT_HOP_ATTR_TUNNEL_ID,
+...
+```
+
+New Bridge-port of type next-hop group is introduced for FDB entries to point to the group of remote VTEPs.
+
+```
+typedef enum _sai_bridge_port_type_t
+...
+    /** Bridge port next hop group */
+    SAI_BRIDGE_PORT_TYPE_BRIDGE_PORT_NEXT_HOP_GROUP,
+} sai_bridge_port_type_t;
+```
+
+The Bridge-port of type next-hop group should have next-hop group ID attribute set.
+```
+typedef enum _sai_bridge_port_attr_t
+{
+...
+    /**
+     * @brief Associated bridge port nexthop group id
+     *
+     * @type sai_object_id_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     * @objects SAI_OBJECT_TYPE_NEXT_HOP_GROUP
+     * @condition SAI_BRIDGE_PORT_ATTR_TYPE == SAI_BRIDGE_PORT_TYPE_BRIDGE_PORT_NEXT_HOP_GROUP
+     */
+    SAI_BRIDGE_PORT_ATTR_BRIDGE_PORT_NEXT_HOP_GROUP_ID,
+...
+}
+```
+
+### 3.2.2 Designated Forwarder support
+
+A new bridgeport attribute defined to drop BUM traffic egressing the bridgeport. The default value of this 
+will be false. 
+Note: Traffic dropped due to this should not be counted against SAI_PORT_STAT_IF_OUT_DISCARDS i.e Interface Tx drop counters.
+
+```
+typedef enum _sai_bridge_port_attr_t
+{
+...
+
+    /**
+     * @brief Indicates if the bridge port is set to drop the Tunnel Terminated broadcast, unknown unicast and multicast traffic
+     * When set to true, egress BUM traffic will be dropped
+     *
+     * @type bool
+     * @flags CREATE_AND_SET
+     * @default false
+     * @validonly SAI_BRIDGE_PORT_ATTR_TYPE == SAI_BRIDGE_PORT_TYPE_PORT or SAI_BRIDGE_PORT_ATTR_TYPE == SAI_BRIDGE_PORT_TYPE_SUB_PORT or SAI_BRIDGE_PORT_ATTR_TYPE == SAI_BRIDGE_PORT_TYPE_TUNNEL

```

Comment:
Do we need bridge port of type "SAI_BRIDGE_PORT_TYPE_TUNNEL" in DF election ? Any use case?
It would be helpful if we can extend "4.3 DF workflow" with bridge port type as SAI_BRIDGE_PORT_TYPE_TUNNEL.

Reply:
There are for "Interconnect ES" (RFC 9014) wherein the ES could be a Tunnel.